src/test/java/com/celements/course/service/CourseServiceTest.java

```diff
@@ -417,33 +421,31 @@ public void test_getConfirmState_stateAbsent() throws Exception {
     addParticipant(regDoc, null);
 
     replayDefault();
-    assertSame(CourseConfirmState.UNDEFINED, courseService.getConfirmState(
+    assertSame(RegistrationState.UNDEFINED, courseService.getConfirmState(
         regDoc.getDocumentReference()));
     verifyDefault();
   }
 
   @Test
-  public void test_getConfirmState_undefinedObj() throws Exception {
+  public void test_getConfirmState_cancelled() throws Exception {
     XWikiDocument regDoc = expectDoc(new DocumentReference("db", "Kurse", "Kurs2"));
-    addParticipant(regDoc, CourseConfirmState.UNCONFIRMED);
-    addParticipant(regDoc, CourseConfirmState.UNDEFINED);
-    addParticipant(regDoc, CourseConfirmState.CONFIRMED);
+    addParticipant(regDoc, RegistrationState.CANCELLED);
 
     replayDefault();
-    assertSame(CourseConfirmState.UNDEFINED, courseService.getConfirmState(
+    assertSame(RegistrationState.CANCELLED, courseService.getConfirmState(
         regDoc.getDocumentReference()));
     verifyDefault();
   }
 
-  private BaseObject addParticipant(XWikiDocument regDoc, CourseConfirmState state) {
+  private BaseObject addParticipant(XWikiDocument regDoc, RegistrationState state) {
     return addParticipant(regDoc, state, null, false);
   }
 
   private BaseObject addParticipant(XWikiDocument regDoc, String email, boolean valid) {
-    return addParticipant(regDoc, CourseConfirmState.UNCONFIRMED, email, valid);
+    return addParticipant(regDoc, RegistrationState.UNCONFIRMED, email, valid);
   }
 
-  private BaseObject addParticipant(XWikiDocument regDoc, CourseConfirmState state, String email,
+  private BaseObject addParticipant(XWikiDocument regDoc, RegistrationState state, String email,
```

Comment:
please fix addParicipant and change RegistrationState parameter to ParticipantStatus type